### ISSUE TITLE
interactive bash

### ISSUE THREAD

**Issue by jcalfee** [mviereck/x11docker#390]
I'm trying to get a simple interactive bash prompt using x11docker version 6.9.0 and:

`$ x11docker --debug --interactive debian:10 bash`

```bash
...
DEBUGNOTE[15:44:34,259]: storeinfo(): containername=x11docker_X112_debian-10-bash_57872207112
DEBUGNOTE[15:44:34,624]: docker command:
  docker run --interactive --tty \
  --name x11docker_X112_debian-10-bash_57872207112 \
  --user 1000:1000 \
  --userns=host \
  --cap-drop ALL \
  --security-opt no-new-privileges \
  --security-opt label=type:container_runtime_t \
  --volume '/usr/bin/docker-init':'/usr/local/bin/init':ro \
  --tmpfs /run:exec --tmpfs /run/lock \
  --volume '/home/james/.cache/x11docker/debian-10-bash-57872207112/share':'/x11docker':rw \
  --volume '/tmp/.X11-unix/X112':'/X112':rw \
  --workdir '/tmp' \
  --entrypoint env \
  --env 'container=docker' \
  --env 'NO_AT_BRIDGE=1' \
  --env 'GTK_CSD=0' \
  --env 'GTK_OVERLAY_SCROLLING=0' \
  --env 'MWWM=allwm' \
  --env 'MWNO_RIT=true' \
  --env 'MWNOCAPTURE=true' \
  --env 'QT_X11_NO_NATIVE_MENUBAR=1' \
  --env 'UBUNTU_MENUPROXY=' \
  --env 'XAUTHORITY=/x11docker/Xauthority.client' \
  --env 'DISPLAY=:112' \
  --env 'USER=james' \
  -- debian:10 /usr/local/bin/init -- /bin/sh - /x11docker/containerrc
DEBUGNOTE[15:44:34,882]: Running xtermrc: Ask for password if needed (no)
DEBUGNOTE[15:44:34,903]: Running dockerrc: Setup as root or as user docker on host.
DEBUGNOTE[15:44:34,904]: storepid(): Stored pid '9016' of 'containershell':  9016 pts/14   00:00:00 bash
DEBUGNOTE[15:44:34,920]: waitforlogentry(): start_xserver(): Waiting for logentry "readyforX=ready" in store.info
DEBUGNOTE[15:44:35,003]: dockerrc: Found default container Runtime: runc
DEBUGNOTE[15:44:35,021]: dockerrc: All  Runtimes: runc
DEBUGNOTE[15:44:35,036]: dockerrc: Container Runtime: UNDECLARED_RUNTIME
DEBUGNOTE[15:44:35,060]: storeinfo(): runtime=UNDECLARED_RUNTIME
DEBUGNOTE[15:44:35,435]: waitforlogentry(): start_xserver(): Stopped waiting for readyforX=ready in store.info due to terminating signal.
DEBUGNOTE[15:44:35,441]: traperror: Command at Line 9257 returned with error code 1:
  return 1
  9271 - ::main::main
DEBUGNOTE[15:44:35,448]: storeinfo(): error=64
DEBUGNOTE[15:44:35,465]: time to say goodbye (traperror)
DEBUGNOTE[15:44:35,490]: waitforlogentry(): xpra: Waiting for logentry "xinitrc=ready" in store.info
DEBUGNOTE[15:44:35,507]: storepid(): Stored pid '9542' of 'xpraloop':  9542 pts/14   00:00:00 bash
DEBUGNOTE[15:44:35,513]: time to say goodbye (main)
DEBUGNOTE[15:44:35,520]: Terminating x11docker.
DEBUGNOTE[15:44:35,524]: time to say goodbye (finish)
DEBUGNOTE[15:44:35,571]: finish(): Checking pid 9542 (xpraloop):  9542 pts/14   00:00:00 bash
DEBUGNOTE[15:44:35,581]: storeinfo(): xtermrc=ready
DEBUGNOTE[15:44:35,587]: waitforlogentry(): start_docker(): Waiting for logentry "dockerrc=ready" in store.info
DEBUGNOTE[15:44:35,604]: termpid(): Terminating 9542 (xpraloop):  9542 pts/14   00:00:00 bash
/usr/bin/x11docker: line 920:  9542 Terminated              { waitforlogentry xpra $Storeinfofile "xinitrc=ready" infinity; rocknroll || exit 0; start_xpra; }
DEBUGNOTE[15:44:35,787]: finish(): Checking pid 9016 (containershell):  9016 pts/14   00:00:00 bash
DEBUGNOTE[15:44:35,836]: termpid(): Terminating 9016 (containershell):  9016 pts/14   00:00:00 bash
/usr/bin/x11docker: line 920:  9016 Terminated              { trap '' SIGINT; case $X11dockermode in 
    run)
        start_docker
    ;;
    exe)
        start_hostexe
    ;;
esac; Pid1pid="$(storeinfo dump pid1pid)"; case $X11dockermode in 
    run)
        case "$Winsubsystem" in 
            "")
                setonwatchpidlist "${Pid1pid:-NOPID}" pid1pid
            ;;
            *)
                setonwatchpidlist "CONTAINER$Containername"
            ;;
        esac
    ;;
    exe)
        setonwatchpidlist "${Pid1pid:-NOPID}" pid1pid
    ;;
esac; case $Xserver in 
    --tty | --hostdisplay | --hostwayland | --weston | --kwin)

    ;;
    *)
        Xinitpid="$(pgrep -a xinit 2>/dev/null | grep "xinit $Xinitrc" | awk '{print $1}')"; checkpid "$Xinitpid" && setonwatchpidlist $Xinitpid xinit; echo $Xcommand | grep -q Xorgwrapper && Line="Xorg $Newdisplay" || Line="$(head -n1 <<< "$Xcommand" | tr -d '\\')"; Xserverpid=$(ps aux | rmcr | grep "$(echo "${Line:-nothingtolookfor}" | cut -d' ' -f1-2)" | grep -v grep | grep -v xinit | awk '{print $2}'); checkpid "$Xserverpid" && setonwatchpidlist "$Xserverpid" Xserver
    ;;
esac; [ "$Pulseaudiomode" = "tcp" ] && start_pulseaudiotcp; checkpid "$Pid1pid" && debugnote "Process tree of ${Hostexe:-container}: (maybe not complete yet)
$(pstree -cp $Pid1pid 2>&1 ||:)"; debugnote "Process tree of x11docker:
$(pstree -p $$ 2>&1 ||:)
  $(storepid test dockerstopshell && echo "Lost child of dockerrc (dockerstopshell):
    $(pstree -p $(storepid dump dockerstopshell) 2>&1 ||:)")"; debugnote "storeinfo(): Stored info:
$(cat $Storeinfofile)"; debugnote "storepid(): Stored pids:
$(cat $Storepidfile)"; [ "$Showinfofile" = "yes" ] && echo "$Storeinfofile"; [ "$Showcache" = "yes" ] && echo "$Cachefolder"; [ "$Showdisplayenvironment" = "yes" ] && echo "$(storeinfo dump Xenv)"; [ "$Showcontainerid" = "yes" ] && echo "$(storeinfo dump containerid)"; [ "$Showcontainerpid1pid" = "yes" ] && echo "$Pid1pid"; storeinfo "x11docker=ready"; } 0<&0
DEBUGNOTE[15:44:36,034]: finish(): Checking pid 8188 (watchmessagefifo):  8188 pts/14   00:00:00 bash
DEBUGNOTE[15:44:36,081]: finish(): Checking pid 8178 (watchpidlist): (already gone)
DEBUGNOTE[15:44:36,153]: Removing container x11docker_X112_debian-10-bash_57872207112
    Error: No such container: x11docker_X112_debian-10-bash_57872207112
DEBUGNOTE[15:44:36,203]: termpid(): Terminating 8188 (watchmessagefifo):  8188 pts/14   00:00:00 bash
/usr/bin/x11docker: line 920:  8188 Terminated              watchmessagefifo
DEBUGNOTE[15:44:36,322]: x11docker exit code: 64
```

**Comment by mviereck** [comment]
Here the command works well, also using xpra.
Your debug output is not complete, it seems to me that the terminating reason appeared above the quoted output.

For comparision, please try with `--tty` instead of automatically chosen option `--xpra`:
```
x11docker --debug --tty --interactive debian:10 bash
```
Please show me a full debug output of a failing command.

**Comment by jcalfee** [comment]
It runs and hangs without dropping into an interactive shell.  This is the full output:

```bash
$ x11docker --debug --tty --interactive debian:10 bash
DEBUGNOTE[16:17:22,106]: check_host(): ps can watch root processes: yes
DEBUGNOTE[16:17:22,145]: host user: james 1000:1000 /home/james
x11docker WARNING: User james is member of group docker.
  That allows unprivileged processes on host to gain root privileges.

DEBUGNOTE[16:17:22,379]: storeinfo(): cache=/home/james/.cache/x11docker/debian-10-bash-59841853899
DEBUGNOTE[16:17:22,389]: storeinfo(): stdout=/home/james/.cache/x11docker/debian-10-bash-59841853899/share/stdout
DEBUGNOTE[16:17:22,398]: storeinfo(): stderr=/home/james/.cache/x11docker/debian-10-bash-59841853899/share/stderr
DEBUGNOTE[16:17:22,421]: storeinfo(): x11dockerpid=15206
DEBUGNOTE[16:17:22,471]: 
x11docker version: 6.9.0
Backend version:   Docker version 19.03.15, build 99e3ed8919
Host system:       "Debian GNU/Linux 9 (stretch)"
Host architecture: amd64 (x86_64)
Command:           '/usr/bin/x11docker' '--debug' '--tty' '--interactive' 'debian:10' 'bash' 
Parsed options:     --debug --tty --interactive -- 'debian:10' 'bash'
DEBUGNOTE[16:17:22,477]: Dependency check for --tty: 0
DEBUGNOTE[16:17:22,481]: Dependencies of --tty already checked: 0 
DEBUGNOTE[16:17:22,490]: storeinfo(): xserver=--tty
DEBUGNOTE[16:17:22,519]: container user: james 1000:1000 /home/james
DEBUGNOTE[16:17:22,573]: storepid(): Stored pid '15705' of 'watchpidlist': 15705 pts/14   00:00:00 bash
DEBUGNOTE[16:17:22,601]: storepid(): Stored pid '15715' of 'watchmessagefifo': 15715 pts/14   00:00:00 bash
DEBUGNOTE[16:17:23,314]: storeinfo(): XDG_RUNTIME_DIR=/run/user/1000
DEBUGNOTE[16:17:23,330]: storeinfo(): Xenv= XDG_RUNTIME_DIR=/run/user/1000
DEBUGNOTE[16:17:23,489]: storeinfo(): tini=/usr/bin/docker-init
DEBUGNOTE[16:17:23,498]: Users and terminal:
  x11docker was started by:                       james
  As host user serves (running X, storing cache): james
  Container user will be:                         james
  Container user password:                        x11docker
  Getting permission to run backend with:         eval 
  Terminal for password frontend:                 bash -c
  Running in a terminal:                          yes
  Running on console:                             no
  Running over SSH:                               no
  Running sourced:                                no
  bash $-:                                        huBE
DEBUGNOTE[16:17:23,505]: storeinfo(): containername=x11docker_X121_debian-10-bash_59841853899
DEBUGNOTE[16:17:23,806]: docker command:
  docker run --interactive --tty \
  --name x11docker_X121_debian-10-bash_59841853899 \
  --user 1000:1000 \
  --userns=host \
  --cap-drop ALL \
  --security-opt no-new-privileges \
  --security-opt label=type:container_runtime_t \
  --volume '/usr/bin/docker-init':'/usr/local/bin/init':ro \
  --tmpfs /run:exec --tmpfs /run/lock \
  --volume '/home/james/.cache/x11docker/debian-10-bash-59841853899/share':'/x11docker':rw \
  --workdir '/tmp' \
  --entrypoint env \
  --env 'container=docker' \
  --env 'WAYLAND_DISPLAY=' \
  --env 'USER=james' \
  -- debian:10 /usr/local/bin/init -- /bin/sh - /x11docker/containerrc
DEBUGNOTE[16:17:24,089]: storepid(): Stored pid '16241' of 'containershell': 16241 pts/14   00:00:00 bash
DEBUGNOTE[16:17:24,110]: Running xtermrc: Ask for password if needed (no)
DEBUGNOTE[16:17:24,131]: waitforlogentry(): start_xserver(): Waiting for logentry "readyforX=ready" in store.info
DEBUGNOTE[16:17:24,144]: Running dockerrc: Setup as root or as user docker on host.
DEBUGNOTE[16:17:24,239]: dockerrc: Found default container Runtime: runc
DEBUGNOTE[16:17:24,263]: dockerrc: All  Runtimes: runc
DEBUGNOTE[16:17:24,294]: dockerrc: Container Runtime: UNDECLARED_RUNTIME
DEBUGNOTE[16:17:24,326]: storeinfo(): runtime=UNDECLARED_RUNTIME
DEBUGNOTE[16:17:24,902]: dockerrc: Image architecture: amd64
DEBUGNOTE[16:17:24,925]: dockerrc: Image USER: 
DEBUGNOTE[16:17:24,943]: storeinfo(): containeruser=james
DEBUGNOTE[16:17:24,973]: dockerrc: Image ENTRYPOINT: 
DEBUGNOTE[16:17:24,987]: dockerrc: Image WORKDIR: 
DEBUGNOTE[16:17:24,998]: storeinfo(): readyforX=ready
DEBUGNOTE[16:17:25,012]: waitforlogentry(): dockerrc: Waiting for logentry "xinitrc is ready" in xinit.log
DEBUGNOTE[16:17:25,157]: waitforlogentry(): start_xserver(): Found log entry "readyforX=ready" in store.info.
DEBUGNOTE[16:17:25,821]: Running xinitrc
DEBUGNOTE[16:17:25,843]: storeinfo(): xinitrc=ready
DEBUGNOTE[16:17:26,041]: waitforlogentry(): dockerrc: Found log entry "xinitrc is ready" in xinit.log.
DEBUGNOTE[16:17:26,065]: storeinfo(): containerid=x11docker_X121_debian-10-bash_59841853899
DEBUGNOTE[16:17:26,282]: dockerrc: Container is up and running.
                                                               DEBUGNOTE[16:17:26,800]: waitforlogentry(): containerrc: Waiting for logentry "containerrootrc=ready" in store.info
                               DEBUGNOTE[16:17:26,828]: dockerrc: 1. check for PID 1: 17015
                                                                                           DEBUGNOTE[16:17:26,855]: storeinfo(): pid1pid=17015
                                                                                                                                              DEBUGNOTE[16:17:26,896]: storeinfo(): containerip=172.17.0.5
                                                       DEBUGNOTE[16:17:26,943]: dockerrc(): Starting containerrootrc with docker exec
                                                                                                                                     DEBUGNOTE[16:17:27,317]: Running containerrootrc: Setup as root in container
                                                              DEBUGNOTE[16:17:27,346]: containerrootrc: Container libc: glibc
                                                                                                                             DEBUGNOTE[16:17:27,422]: containerrootrc: james:x:1000:1000:james,,,:/home/james:/bin/bash
                                                                    DEBUGNOTE[16:17:27,444]: storeinfo(): containerrootrc=ready
                                                                                                                               DEBUGNOTE[16:17:27,454]: waitforlogentry(): start_docker(): Waiting for logentry "dockerrc=ready" in store.info
                                                                                           DEBUGNOTE[16:17:27,462]: storeinfo(): dockerrc=ready
                                                                                                                                               DEBUGNOTE[16:17:27,466]: waitforlogentry(): start_docker(): Found log entry "dockerrc=ready" in store.info.
                                                                                                       DEBUGNOTE[16:17:27,480]: storepid(): Stored pid '17337' of 'dockerstopshell': 17337 pts/14   00:00:00 bash
                                                              DEBUGNOTE[16:17:27,483]: watchpidlist(): Setting pid 17015 on watchlist: pid1pid
                                                                                                                                              DEBUGNOTE[16:17:27,502]: storeinfo(): xtermrc=ready
                                              DEBUGNOTE[16:17:27,539]: watchpidlist(): Watching pids: 
                                                                                                      17015 pts/0    00:00:00 init
                                                                                                                                  DEBUGNOTE[16:17:27,552]: storepid(): Stored pid '17015' of 'pid1pid': 17015 pts/0    00:00:00 init
                                                                                 DEBUGNOTE[16:17:27,670]: Process tree of container: (maybe not complete yet)
          init(17015)---sh(17054)---sleep(17234)
                                                DEBUGNOTE[16:17:27,812]: waitforlogentry(): containerrc: Found log entry "containerrootrc=ready" in store.info.
            DEBUGNOTE[16:17:27,839]: Running containerrc: Unprivileged user commands in container
                                                                                                 DEBUGNOTE[16:17:27,925]: Process tree of x11docker:
 bash(15206)-+-bash(15697)---tail(15700)
                                                    |-bash(15705)
                                                                             |-bash(15715)
                                                                                                      |-bash(16241)---bash(17483)---pstree(17484)
                                                                                                                                                             `-bash(16822)
                         Lost child of dockerrc (dockerstopshell):
                                                                      bash(17337)
                                                                                 DEBUGNOTE[16:17:27,925]: containerrc: HOME is empty. Copying from /etc/skel
         DEBUGNOTE[16:17:27,934]: storeinfo(): Stored info:
                                                           cache=/home/james/.cache/x11docker/debian-10-bash-59841853899
                                                                                                                        stdout=/home/james/.cache/x11docker/debian-10-bash-59841853899/share/stdout
                                                stderr=/home/james/.cache/x11docker/debian-10-bash-59841853899/share/stderr
                                                                                                                           x11dockerpid=15206
                                                                                                                                             xserver=--tty
       XDG_RUNTIME_DIR=/run/user/1000
                                     Xenv= XDG_RUNTIME_DIR=/run/user/1000
                                                                         tini=/usr/bin/docker-init
                                                                                                  containername=x11docker_X121_debian-10-bash_59841853899
      runtime=UNDECLARED_RUNTIME
                                containeruser=james
                                                   readyforX=ready
                                                                  xinitrc=ready
                                                                               containerid=x11docker_X121_debian-10-bash_59841853899
                                                                                                                                    pid1pid=17015
                                                                                                                                                 containerip=172.17.0.5
                    containerrootrc=ready
                                         dockerrc=ready
                                                       xtermrc=ready
                                                                    DEBUGNOTE[16:17:27,947]: storepid(): Stored pids:
                                                                                                                     15705 watchpidlist
                                                                                                                                       15715 watchmessagefifo
          16241 containershell
                              17337 dockerstopshell
                                                   17015 pid1pid
                                                                DEBUGNOTE[16:17:27,958]: storeinfo(): x11docker=ready


```

**Comment by mviereck** [comment]
Oh, I just see that there has been a bug for interactive in v6.9.0. It is fixed in recent stable version v6.10.0. 
Please update first.
